ansible/roles/basic_users/tasks/main.yml

```diff
@@ -0,0 +1,44 @@
+---
+- name: Terminate systemd user sessions for absent users
+  terminate_user_sessions:
+    user: "{{ item.name }}"
+  loop: "{{ basic_users_users }}"
+  loop_control:
+    label: "{{ item.name }}"
+  when:
+    - "item.state | default('present') == 'absent'"
+  
+- name: Create users and generate public keys
+  user: "{{ basic_users_userdefaults | combine(item) | remove_keys(['public_key']) }}"
+  loop: "{{ basic_users_users }}"
+  loop_control:
+    label: "{{ item.name }} [{{ item.state | default('present') }}]"
+  register: basic_users_info
+
+- name: Write supplied public key as authorized for SSH access
+  authorized_key:
+    user: "{{ item.name }}"
+    state: present
+    key: "{{ item.public_key }}"
+  loop: "{{ basic_users_users }}"
+  loop_control:
+    label: "{{ item.name }} [{{ item.state | default('present') }}]"
+  when:
+    - item.state | default('present') == 'present'
+    - item.public_key is defined
+    - basic_users_manage_homedir
+  run_once: true
+
+- name: Write generated public key as authorized for SSH access
+  authorized_key:
+    user: "{{ item.name }}"
+    state: present
+    manage_dir: no
+    key: "{{ item.ssh_public_key }}"
+  loop: "{{ hostvars[ansible_play_hosts | first].basic_users_info.results }}"
```

Comment:
I think `basic_users_info` is set as a fact on all hosts, right?

Reply:
Yes but the key is only generated on the first one (by design), so it is only present in the facts for that host.